Remove the unused AM_C_PROTOTYPES call from configure.ac; fixing FTBFS with
newer autoconf.

http://dev.mutt.org/trac/changeset/0488deb39a35
http://dev.mutt.org/trac/ticket/3579

Index: b/configure.ac
===================================================================
--- a/configure.ac
+++ b/configure.ac
@@ -28,7 +28,6 @@
 
 AC_PROG_CC
 AC_ISC_POSIX
-AM_C_PROTOTYPES
 if test "x$U" != "x"; then
   AC_MSG_ERROR(Compiler not ANSI compliant)
 fi
